WordPress/src/main/java/org/wordpress/android/sharedlogin/resolver/LocalMigrationOrchestrator.kt

```diff
@@ -3,108 +3,73 @@ package org.wordpress.android.sharedlogin.resolver
+    private val sharedLoginHelper: SharedLoginHelper,
 ) {
     fun tryLocalMigration() {
         localMigrationContentResolver.getResultForEntityType<EligibilityStatusData>(EligibilityStatus).validate()
-                .then {
-                    originalTryLocalMigration()
+                .then(sharedLoginHelper::login)
+                .thenWith {
+                    originalTryLocalMigration(it.token)
                     Success(it)
-                }.otherwise(::handleErrors)
+                }
+                .otherwise(::handleErrors)
     }
 
     @Suppress("ForbiddenComment")
     // TODO: Handle the errors appropriately
     private fun handleErrors(error: LocalMigrationError) {
         when(error) {
             is ProviderError -> Unit
-            is Ineligibility -> Unit
+            is Ineligibility -> when (error.reason) {
+                WPNotLoggedIn -> sharedLoginAnalyticsTracker.trackLoginFailed(ErrorType.WPNotLoggedInError)
+            }
+            is FeatureDisabled -> Unit
+            is MigrationAlreadyAttempted -> Unit
         }
     }
-    private fun originalTryLocalMigration() {
-        @Suppress("ForbiddenComment")
-        // TODO: We should move this login specific logic to a helper. It can happen as a later step, since the
-        // eligibility check is handled separately now (on the provider side).
-        val isFeatureFlagEnabled = jetpackSharedLoginFlag.isEnabled()
-
-        if (!isFeatureFlagEnabled) {
-            return
-        }
-
-        val hasSelfHostedSites = siteStore.hasSite() && siteStore.sites.any { !it.isUsingWpComRestApi }
-        val isAlreadyLoggedIn = accountStore.hasAccessToken() || hasSelfHostedSites
-        val isFirstTry = appPrefsWrapper.getIsFirstTrySharedLoginJetpack()
-
-        if (isAlreadyLoggedIn || !isFirstTry) {
-            return
-        }
-
-        sharedLoginAnalyticsTracker.trackLoginStart()
-        appPrefsWrapper.saveIsFirstTrySharedLoginJetpack(false)
-        val (accessToken) = localMigrationContentResolver.getDataForEntityType<AccessTokenData>(AccessToken)
-        val (sites) = localMigrationContentResolver.getDataForEntityType<SitesData >(Sites)
-        val hasLocalSelfHostedSites = sites.any { !it.isUsingWpComRestApi }
+    private fun originalTryLocalMigration(accessToken: String) {
         @Suppress("ForbiddenComment")
-        // TODO: Unify error tracking for resolver / provider errors too
-        if (accessToken.isNotEmpty() || hasLocalSelfHostedSites) {
-            runFlow(accessToken, sites)
-        } else {
-            sharedLoginAnalyticsTracker.trackLoginFailed(ErrorType.WPNotLoggedInError)
-        }
-    }
-
-    private fun runFlow(accessToken: String, sites: List<SiteModel>) {
-        val hasSites = sites.isNotEmpty()
-
-        if (hasSites) {
-            resolverUtility.copySitesWithIndexes(sites)
-        }
-
-        sharedLoginAnalyticsTracker.trackLoginSuccess()
+        // TODO: Extract sites migration to helper
```

Comment:
Should this `TODO` be handled in this PR or it will be handled in another one?

Reply:
My understanding is that [this is handled in the next PR](https://github.com/wordpress-mobile/WordPress-Android/pull/17484).

Reply:
That's right, this is tackled in a later PR in the "chain" :smile: